### DISCUSSION TITLE
How to use 'orderBy' on a joined query?

### DISCUSSION THREAD

**Discussion by mrverdant13** [simolus3/drift#1045] [Q&A]
When I use the following expresion, an exception is thrown.

```dart
final joinedQuery = select(
      firstTable,
    ).join(
      [
        innerJoin(
          secondTable,
          secondTable.id.equalsExp(
            firstTable.secondId,
          ),
        ),
      ],
    );

    joinedQuery
      ..where( ... )
      ..orderBy(
        [OrderingTerm.asc(secondTable.name)],
      );

    return joinedQuery.get()
```

![image](https://user-images.githubusercontent.com/42245236/107728063-0005db00-6cbb-11eb-832b-92bcf62e3701.png)

**Comment by simolus3** [comment]
Does the I/O error only happen for that query? From a quick search, it looks like that error could happen if an auxiliary journey file for sqlite3 isn't writable. I don't really see how this condition could be triggered from a specific query, do you have a repro for me to look at?

**Reply by mrverdant13** [reply]
It only happen for that specific query, yes.
My workaround was to sort the results inline (with Dart) but I am aware of this as a bad approach.
I have no public repo for reference but I am going to set up a new one with a similar query.
Must point this out: the issue shows up when the `orderBy` caluse is appended that way (after a `join` set).
